InvenTree/company/migrations/0035_supplierpart_update.py

```diff
@@ -0,0 +1,95 @@
+import InvenTree.fields
+from django.db import migrations, models, transaction
+import django.db.models.deletion
+from django.db.utils import IntegrityError
+
+def supplierpart_make_manufacturer_parts(apps, schema_editor):
+    Part = apps.get_model('part', 'Part')
+    ManufacturerPart = apps.get_model('company', 'ManufacturerPart')
+    SupplierPart = apps.get_model('company', 'SupplierPart')
+
+    supplier_parts = SupplierPart.objects.all()
+    
+    if supplier_parts:
+        print(f'\nCreating Manufacturer parts\n{"-"*10}')
+        for supplier_part in supplier_parts:
+            print(f'{supplier_part.supplier.name[:15].ljust(15)} | {supplier_part.SKU[:15].ljust(15)}\t', end='')
+
+            if supplier_part.manufacturer_part:
+                print(f'[ERROR: MANUFACTURER PART ALREADY EXISTS]')
+                continue
+
+            part = supplier_part.part
+            if not part:
+                print(f'[ERROR: SUPPLIER PART IS NOT CONNECTED TO PART]')
+                continue
+            
+            manufacturer = supplier_part.manufacturer
+            MPN = supplier_part.MPN
+            link = supplier_part.link
+            description = supplier_part.description
+
+            if manufacturer or MPN:
+                print(f' | {part.name[:15].ljust(15)}', end='')
+                
+                try:
+                    print(f' | {manufacturer.name[:15].ljust(15)}', end='')
+                except AttributeError:
+                    print(f' | {"EMPTY MANUF".ljust(15)}', end='')
+
+                try:
+                    print(f' | {MPN[:15].ljust(15)}', end='')
+                except TypeError:
+                    print(f' | {"EMPTY MPN".ljust(15)}', end='')
+
+                print('\t', end='')
+
+                # Create ManufacturerPart
+                manufacturer_part = ManufacturerPart(part=part, manufacturer=manufacturer, MPN=MPN, description=description, link=link)
+                created = False
+                try:
+                    with transaction.atomic():
+                        manufacturer_part.save()
+                    created = True
+                except IntegrityError:
+                    manufacturer_part = ManufacturerPart.objects.get(part=part, manufacturer=manufacturer, MPN=MPN)
+
+                # Link it to SupplierPart
+                supplier_part.manufacturer_part = manufacturer_part
+                supplier_part.save()
+
+                if created:
+                    print(f'[SUCCESS: MANUFACTURER PART CREATED]')
+                else:
+                    print(f'[IGNORED: MANUFACTURER PART ALREADY EXISTS]')
+            else:
+                print(f'[IGNORED: MISSING MANUFACTURER DATA]')
+
+        print(f'{"-"*10}\nDone\n')
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('company', '0033_auto_20210410_1528'),
+        ('company', '0034_manufacturerpart'),
+    ]
+
+    operations = [
```

Comment:
The data migration looks really good, fantastic work!

A couple of points on the `migrations.RunPython` code (below)

- You should add a `reverse_code` function to a `migrations.RunPython()` call - so that the migration can be reversed if necessary
- Have a look at other instances of `migrations.RunPython` in the code base

Reply:
Done!